pulsar/consumer_impl.go

```diff
@@ -589,11 +589,20 @@ func (c *consumer) Seek(msgID MessageID) error {
 func (c *consumer) SeekByTime(time time.Time) error {
 	c.Lock()
 	defer c.Unlock()
-	if len(c.consumers) > 1 {
-		return newError(SeekFailed, "for partition topic, seek command should perform on the individual partitions")
+	errorSeek := make([]error, len(c.consumers))
```

Comment:
Should return all errors, rather than just first. pulsar-client-go already uses `pkgerrors "github.com/pkg/errors"`
```
var err error

for i, cons := range c.consumers {
	if err := cons.SeekByTime(time); err != nil {
		// create as appropriate - msg := fmt.Sprintf("unable to SeekByTime for topic=%s partition=%s", c.Topic(), c.Subscription())
		errs = pkgerrors.Wrap(err, msg)
	}
}

return errs
```

pulsar-client-go example - https://github.com/apache/pulsar-client-go/blob/5108332c9dd4cb454c26804304bffb82eeffc713/pulsar/consumer_multitopic.go#L88-L98
pkgerrors.Wrap implementation - https://github.com/pkg/errors/blob/master/errors.go#L181-L196